subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JfrProfiler.groovy

```diff
@@ -90,24 +89,34 @@ class JfrProfiler extends Profiler implements Stoppable {
         pid.gradleArgs
     }
 
-    private File getJfrFile(BuildExperimentSpec spec) {
+    private File getJfrOutputDirectory(BuildExperimentSpec spec) {
         def fileSafeName = spec.displayName.replaceAll('[^a-zA-Z0-9.-]', '-').replaceAll('-+', '-')
         def baseDir = new File(logDirectory, fileSafeName)
-        new File(baseDir, "profile.jfr")
+        def outputDir = new File(baseDir, "jfr-recordings")
+        outputDir.mkdirs()
+        return outputDir
     }
 
     void start(BuildExperimentSpec spec) {
+        // Remove any profiles created during warmup
+        def jfrOutputDir = getJfrOutputDirectory(spec)
```

Comment:
If you want to clean the whole folder, consider something like:
```groovy
FileUtils.cleanDirectory(getJfrOutputDirectory(spec))
```
or if you want to filter, maybe consider condensing it into:
```groovy
getJfrOutputDirectory(spec).eachFileMatch(~/.+\.jfr/) { it.delete() }
```

Reply:
Thanks, I changed it.